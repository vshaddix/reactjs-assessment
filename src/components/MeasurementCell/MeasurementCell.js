import React from "react";
import './MeasurementCell.scss';
import LoadingContent from '../LoadingContent/LoadingContent';

export default ({ parameter, unit, value }) => {
  if (!parameter) {
    return <LoadingContent linesCount={1} />;
  }

  const displayValue = !value ? (<span>-</span>) : (<span>{value} {unit}</span>);

  return (
    <div>
      <span>{parameter}</span>: {displayValue}
    </div>
  );
};
