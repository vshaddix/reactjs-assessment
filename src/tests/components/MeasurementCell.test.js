import React from 'react';
import { create } from "react-test-renderer";
import MeasurementCell from '../../components/MeasurementCell/MeasurementCell';

it('renders (snapshot test)', () => {
  const component = create(<MeasurementCell />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when passed full data', () => {
  const component = create(<MeasurementCell parameter="test" unit="test-unit" value={21} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders the component correctly when no value is passed', () => {
  const component = create(<MeasurementCell parameter="test" unit="test-unit" />);

  expect(component.toJSON()).toMatchSnapshot();
});

