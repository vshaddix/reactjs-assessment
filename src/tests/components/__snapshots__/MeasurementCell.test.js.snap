// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `
<div
  className="ui placeholder"
>
  <div
    className="line"
  />
</div>
`;

exports[`renders the component correctly when no value is passed 1`] = `
<div>
  <span>
    test
  </span>
  : 
  <span>
    -
  </span>
</div>
`;

exports[`renders the component correctly when passed full data 1`] = `
<div>
  <span>
    test
  </span>
  : 
  <span>
    21
     
    test-unit
  </span>
</div>
`;
